controllers/flinkcluster_observer.go

```diff
@@ -294,13 +296,16 @@ func (observer *ClusterStateObserver) observeFlinkJobStatus(observed *ObservedCl
 	flinkJobStatus.flinkJobList = flinkJobList
 
 	// Check running jobs
-	if len(flinkJobList.Jobs) < 1 {
+	if len(flinkJobList.Jobs) < 1 && observed.flinkJobSubmitLog == nil {
 		return
 	}
 
-	flinkJobStatus.flinkJob = &flinkJobList.Jobs[0]
-	for _, job := range flinkJobList.Jobs[1:] {
-		flinkJobStatus.flinkJobsUnexpected = append(flinkJobStatus.flinkJobsUnexpected, job.Id)
+	for _, job := range flinkJobList.Jobs {
+		if observed.flinkJobSubmitLog.JobID == job.Id {
+			flinkJobStatus.flinkJob = &job
```

Comment:
@regadas null pointer exception can happen here when the job pod is lost

Reply:
Nice catch! indeed it misses a check previously.